test_data/rules/unrelated_type_equality_checks.dart

```diff
@@ -203,3 +217,6 @@ class ClassWCall {
 }
 
 class SubClassWCall extends ClassWCall {}
+
+enum EnumOpKind1 { insert, update, delete }
```

Comment:
It shouldn't affect the results, but I'd like a few more enum test cases. For example, the following are legal, and should still be considered unrelated to other enums: `enum E implements C` and `enum E with M`.